src/cdk/layout/breakpoints-observer.ts

```diff
@@ -0,0 +1,100 @@
+import {Subject} from 'rxjs/Subject';
+import {RxChain, map, startWith, takeUntil} from '@angular/cdk/rxjs';
+import {coerceArray} from '@angular/cdk/coercion';
+import {combineLatest} from 'rxjs/observable/combineLatest';
+import {fromEventPattern} from 'rxjs/observable/fromEventPattern';
+
+export type BreakpointQuery = string | string[];
+
+export interface BreakpointState {
+  query: string;
+  matches: boolean;
+}
+
+interface Query {
+  observable: Observable<BreakpointState>;
+  mql: MediaQueryList;
+}
+
+@Injectable()
+export class BreakpointObserver implements OnDestroy {
+  // A map of all media queries currently being listened for.
+  private _queries: Map<string, Query> = new Map();
+  // A subject for all other observables to takeUntil based on.
+  private _activeSubject: Subject<{}> = new Subject();
+
+  constructor(private mediaMatcher: MediaMatcher, private zone: NgZone) {}
+
+  /** Completes the active subject, signalling to all other observables to complete. */
+  ngOnDestroy() {
+    this._activeSubject.next();
+    this._activeSubject.complete();
+  }
+
+  /**
+   * Whether the query currently is matched.
+   */
+  isMatched(value: BreakpointQuery): boolean {
+    let queries = coerceArray(value);
+    return queries.some((mediaQuery: string) => {
+      return this._registerQuery(mediaQuery).mql.matches;
+    });
+  }
+
+  /**
+   * Retrieves an observable for the BreakpointQuery, broadcasting changes to the matching state.
+   */
+  observe(value: BreakpointQuery): Observable<BreakpointState> {
+    let queries = coerceArray(value);
+    let observables = queries.map(query => this._registerQuery(query).observable);
+
+    return combineLatest(observables, (a: BreakpointState, b: BreakpointState) => {
+      return <BreakpointState>{
+        matches: (a && a.matches) || (b && b.matches) || false,
+        query: queries.join(', ')
+      };
+    });
+  }
+
+  /**
+   * Registers a specific query to be listened for.
+   */
+  private _registerQuery(query: string): Query {
+    // Only set up a new MediaQueryList if it is not already being listened for.
+    if (this._queries.has(query)) {
+      return this._queries.get(query)!;
+    }
+    // Create new MediaQueryList.
+    let mql: MediaQueryList = this.mediaMatcher.matchMedia(query);
+    // Create callback for match changes and add it is as a listener.
+    let queryObservable = RxChain.from(fromEventPattern(
+      // Listener callback methods are wrapped to be placed back in ngZone.
+      (listener: MediaQueryListListener) => {
+        mql.addListener((e: MediaQueryList) => this.zone.run(() => listener(e)));
+      },
+      (listener: MediaQueryListListener) => {
+        mql.removeListener((e: MediaQueryList) => this.zone.run(() => listener(e)));
+      }))
+      .call(takeUntil, this._activeSubject)
+      .call(startWith, mql)
+      .call(map, (nextMql: MediaQueryList) => {
+        return <BreakpointState>{query: query, matches: nextMql.matches};
```

Comment:
I think you don't need this type coercion. Without it, the operator fits on one line
```ts
.call(map, (nextMql: MediaQueryList) => ({query: query, matches: nextMql.matches}))
```